source/m2mnsdlinterface.cpp

```diff
@@ -1683,3 +1727,27 @@ void M2MNsdlInterface::change_operation_mode(M2MObject *object, M2MBase::Operati
         }
     }
 }
+
+void M2MNsdlInterface::set_server_address(const char* server_address)
+{
+    free(_server_address);
+    _server_address = M2MBase::alloc_string_copy(server_address);
+}
+
+bool M2MNsdlInterface::parse_and_send_uri_query_parameters()
+{
+    char *address_copy = M2MBase::alloc_string_copy(_server_address);
+    char* query = query_string((const char*)_server_address);
+    bool msg_sent = false;
+    if (query != NULL) {
+        int param_count = query_param_count(query);
+        char* uri_query_params[param_count];
```

Comment:
Shouldn't the param count be verified for zero? That can happen if there is no query parameter. And there should also be a unit test for that case.